### PR TITLE
ArduCopter : added and fixed few comments

### PR DESCRIPTION
**Pull request by shiv-tyagi**
Hi,
I have just started to contribute to open source. Yesterday, I created a pull request for the same thing but there I forgot to squash few of my commits into one, and when asked, while squashing them I got few conflicts and that messed up the things. Hence, I closed that PR and here is my fresh PR with a single commit. I've tried keeping things mentioned in contributing documentation (like rebasing etc.) in mind. I hope I did it correctly this time.

### CODE REVIEW

**Comment by rmackay9** [comment]
Looks pretty OK.  Could you change the commit message to start with "Copter: "?

So the commit message could be, "Copter: added comments to include libraries in copter.h and attitude.cpp"

File: ArduCopter/Copter.h [hunk 1]
```diff
@@ -28,45 +28,45 @@
...
+#include <AP_BattMonitor/AP_BattMonitor.h>  // Battery monitor library
+#include <AP_LandingGear/AP_LandingGear.h>  // Landing Gear library
 #include <AC_InputManager/AC_InputManager.h>        // Pilot input handling library
 #include <AC_InputManager/AC_InputManager_Heli.h>   // Heli specific pilot input handling library
 #include <AP_Arming/AP_Arming.h>
 #include <AP_SmartRTL/AP_SmartRTL.h>
-#include <AP_TempCalibration/AP_TempCalibration.h>
-#include <AC_AutoTune/AC_AutoTune.h>
+#include <AP_TempCalibration/AP_TempCalibration.h>  //temperature calibration library
+#include <AC_AutoTune/AC_AutoTune.h>        //ardupilot autotune library. support for autotune of multirotors.
```
Comment (Hwurzburg) [inline ArduCopter/Copter.h:66]: nitpick....we always use ArduPilot rather than ardupilot

**Review by Hwurzburg — CHANGES_REQUESTED** [review]

**Comment by shiv-tyagi** [comment]
> Looks pretty OK. Could you change the commit message to start with "Copter: "?
> 
> So the commit message could be, "Copter: added comments to include libraries in copter.h and attitude.cpp"
> 
> If that's too difficult I can do it for you probably.

I tried renaming the commit message with git interactive rebase. But when I push the renamed commit, it says 'Updates were rejected because the tip of your current branch is behind'. I guess this is happening because I am renaming an already pushed commit. Can you please guide me what to do in this case? @rmackay9

**Comment by rmackay9** [comment]
@shiv-tyagi, maybe try using "git gui" and then you can amend the previous commit.

It should be possible with an interactive rebase as well,  change the first few characters on the line for the commit to "r" and then  save and exit the editor.

**Comment by shiv-tyagi** [comment]
@rmackay9 I tried 'git gui' but am not getting an option to amend that commit. All I can do is reverting that commit (that would add another commit). Should I do another commit with new message? Will that work?

**Comment by hendjoshsr71** [comment]
@shiv-tyagi Once your local commit message & code is correct now, ie "Copter: added comments to include libraries in copter.h and attitude.cpp"

What you need to do is force push the branch as you want to overwrite the commits. I've only used GIT GUI once or twice, but I believe you want to check the box, "Force overwrite ..."

**Comment by shiv-tyagi** [comment]
@hendjoshsr71 Hi Josh, Thanks for helping me out with this. I just force pushed my branch. I hope I did it correctly. It feels really good to learn new things. I'll definitely check out the course suggested by you. Thanks again!!

File: ArduCopter/Copter.h [hunk 2]
```diff
@@ -28,45 +28,45 @@
...
+#include <AP_Declination/AP_Declination.h>  // ArduPilot Mega Declination Helper Library
 #include <AP_RCMapper/AP_RCMapper.h>        // RC input mapping library
-#include <AP_BattMonitor/AP_BattMonitor.h>     // Battery monitor library
-#include <AP_LandingGear/AP_LandingGear.h>     // Landing Gear library
+#include <AP_BattMonitor/AP_BattMonitor.h>  // Battery monitor library
+#include <AP_LandingGear/AP_LandingGear.h>  // Landing Gear library
 #include <AC_InputManager/AC_InputManager.h>        // Pilot input handling library
 #include <AC_InputManager/AC_InputManager_Heli.h>   // Heli specific pilot input handling library
 #include <AP_Arming/AP_Arming.h>
 #include <AP_SmartRTL/AP_SmartRTL.h>
```
Comment (hendjoshsr71) [inline ArduCopter/Copter.h:64]: You could add comments to these ones as well.

File: ArduCopter/Copter.h [hunk 3]
```diff
@@ -28,45 +28,45 @@
...
-#include <AP_TempCalibration/AP_TempCalibration.h>
-#include <AC_AutoTune/AC_AutoTune.h>
+#include <AP_TempCalibration/AP_TempCalibration.h>  //temperature calibration library
+#include <AC_AutoTune/AC_AutoTune.h>        //ArduPilot autotune library. support for autotune of multirotors.
 #include <AP_Parachute/AP_Parachute.h>
```
Comment (hendjoshsr71) [inline ArduCopter/Copter.h:67]: comment here too

File: ArduCopter/Copter.h [hunk 4]
```diff
@@ -28,45 +28,45 @@
...
-#include <AC_Sprayer/AC_Sprayer.h>
-#include <AP_ADSB/AP_ADSB.h>
+#include <AC_Sprayer/AC_Sprayer.h>          //Crop sprayer library
+#include <AP_ADSB/AP_ADSB.h>                //ADS-B RF based collision avoidance module library
 #include <AP_Proximity/AP_Proximity.h>
```
Comment (hendjoshsr71) [inline ArduCopter/Copter.h:70]: comment here

**Review by hendjoshsr71 — COMMENTED** [review]
Since you are adding comments. You may as well add comments to describe the last few `#includes`.

**Comment by hendjoshsr71** [comment]
> @hendjoshsr71 Hi Josh, Thanks for helping me out with this. I just force pushed my branch. I hope I did it correctly. It feels really good to learn new things. I'll definitely check out the course suggested by you. Thanks again!!

Looking much better! 

By the way here is the course I mentioned. https://www.coursera.org/lecture/version-control-with-git/git-overview-WFdJp  Note: that ArduPilot doesn't use merge commits, but other projects do. So they will go over merging a lot in it. 

If you are interested in making some functional changes the "Good First Issue" tag is good place to start. 
https://github.com/ArduPilot/ardupilot/labels/good%20first%20issue Just ask on the issue if you need clarification about it.

**Comment by shiv-tyagi** [comment]
Sure. Actually, as I am beginning my open source journey with ArduPilot, I wanted to learn about this code base. So, I felt that adding comments along with learning about the code base will help me learn fast and also make me familiar to contribution process. The libraries against which I have not commented are the ones about which I could not figure out what they are exactly for. And I did not want to write any wrong information in comments. Anyway, I'll try finding out for what they are and it would be great if you can tell me a bit about these. That would make me understand them better.

**Comment by hendjoshsr71** [comment]
Totally understandable!

The library names should be self describing or give you at least an idea of they do before you go into the code. Also don't forget about the documentation on the wiki both on the developer and vehicle specific versions. 

For example google, "ardupilot smart rtl" you can find https://ardupilot.org/copter/docs/smartrtl-mode.html

File: ArduCopter/Copter.h (hunk 2 above)
Comment (shiv-tyagi) [reply ArduCopter/Copter.h]: Done!

File: ArduCopter/Copter.h (hunk 3 above)
Comment (shiv-tyagi) [reply ArduCopter/Copter.h]: Done!

File: ArduCopter/Copter.h (hunk 4 above)
Comment (shiv-tyagi) [reply ArduCopter/Copter.h]: Done

**Pull request merged**